plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -221,6 +228,14 @@ public IcebergMetadata(
         this.commitTaskCodec = requireNonNull(commitTaskCodec, "commitTaskCodec is null");
         this.tableOperationsProvider = requireNonNull(tableOperationsProvider, "tableOperationsProvider is null");
         this.trinoVersion = requireNonNull(trinoVersion, "trinoVersion is null");
+        this.tableCache = CacheBuilder.newBuilder()
+                .maximumSize(100)
+                .expireAfterWrite(5, TimeUnit.MINUTES)
```

Comment:
The IcebergMetadata object is per-query scoped, so you don't need time-based eviction here.
That's why `tableMetadataCache` uses ordinary map.

Reply:
Ah, I did not know that. Then we do not even need the two levels query -> table but one level is enough.